vm/src/gas/price_list.rs

```diff
@@ -0,0 +1,192 @@
+    /// the base is covering for the put.
+    pub create_actor_base: i64,
+    pub create_actor_extra: i64,
+
+    /// Gas cost for deleting an actor.
+    /// Note: this partially refunds the create cost to incentivise the deletion of the actors.
+    pub delete_actor: i64,
+
+    pub hashing_base: i64,
+    pub hashing_per_byte: i64,
+
+    pub compute_unsealed_sector_cid_base: i64,
+    pub verify_seal_base: i64,
+    pub verify_post_base: i64,
+    pub verify_consensus_fault: i64,
+}
+
+impl PriceList {
+    /// Returns the gas required for storing a message of a given size in the chain.
+    #[inline]
+    pub fn on_chain_message(&self, msg_size: i64) -> i64 {
+        self.on_chain_message_base + self.on_chain_message_per_byte * msg_size
+    }
+    /// Returns the gas required for storing the response of a message in the chain.
+    #[inline]
+    pub fn on_chain_return_value(&self, data_size: usize) -> i64 {
+        data_size as i64 * self.on_chain_return_value_per_byte
+    }
+    /// Returns the gas required when invoking a method.
+    #[inline]
+    pub fn on_method_invocation(&self, value: &TokenAmount, method_num: MethodNum) -> i64 {
+        let mut ret = self.send_base;
+        if value != &TokenAmount::zero() {
+            ret += self.send_transfer_funds;
+        }
+        if method_num != METHOD_SEND {
+            ret += self.send_invoke_method;
+        }
+        ret
+    }
+    /// Returns the gas required for storing an object
+    #[inline]
+    pub fn on_ipld_get(&self, data_size: usize) -> i64 {
+        self.ipld_get_base + data_size as i64 * self.ipld_get_per_byte
+    }
+    /// Returns the gas required for storing an object
+    #[inline]
+    pub fn on_ipld_put(&self, data_size: usize) -> i64 {
+        self.ipld_put_base + data_size as i64 * self.ipld_put_per_byte
+    }
+    /// Returns the gas required for creating an actor
+    #[inline]
+    pub fn on_create_actor(&self) -> i64 {
+        self.create_actor_base + self.create_actor_extra
+    }
+    /// Returns the gas required for deleting an actor
+    #[inline]
+    pub fn on_delete_actor(&self) -> i64 {
+        self.delete_actor
+    }
+    /// Returns gas required for signature verification
+    #[inline]
+    pub fn on_verify_signature(&self, sig_type: SignatureType, plain_text_size: usize) -> i64 {
+        match sig_type {
+            SignatureType::BLS => (3 * plain_text_size + 2) as i64,
+            SignatureType::Secp256 => (3 * plain_text_size + 2) as i64,
+        }
+    }
+    /// Returns gas required for hashing data
+    #[inline]
+    pub fn on_hashing(&self, data_size: usize) -> i64 {
+        self.hashing_base + data_size as i64 * self.hashing_per_byte
+    }
+    /// Returns gas required for computing unsealed sector Cid
+    #[inline]
+    pub fn on_compute_unsealed_sector_cid(
+        &self,
+        _proof: RegisteredProof,
+        _pieces: &[PieceInfo],
+    ) -> i64 {
+        self.compute_unsealed_sector_cid_base
```

Comment:
Nit: add a TODO here

Reply:
There is nothing else TODO for all of these, but I will add a comment to revisit as this probably will change in future

Reply:
I figured a TODO must be required since we are not using the params?